### ISSUE TITLE
Add option to (Adaptive)SamplingTelemetryProcessor to not write filtered items to Debug output

### ISSUE THREAD

**Issue by EnCey** [microsoft/ApplicationInsights-dotnet#968]
##  Actual Behavior
The [SamplingTelemetryProcessor](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/91e9c91fcea979b1eec4e31ba8e0fc683bf86802/src/ServerTelemetryChannel/SamplingTelemetryProcessor.cs#216) writes sampled telemetry items to the `TelemetryDebugWriter` i.e. the Debug output window.

Sampled items thus appear not only in the output window, but also AI search.

## Expected Behavior
There should be an option to not write sampled items to Debug.WriteLine

## Use case
I created a custom `TelemetryProcessor` that filters out noisy telemetry items in local development builds. Stuff like metrics or successful static file requests from ASP.Net Core.

The problem is that this processor comes after the `SamplingTelemetryProcessor` (and there is no way to affect that order?), which may choose to "sample" these unwanted telemetry items and thus make them appear in the output window.

In this specific case, we have a debug build of a web app that requests nearly 1000 small files (mostly `.js`). When all these requests are logged, it absolutely cripples the ASP.Net server and a cache refresh takes nearly a minute to complete. 

The custom `TelemetryProcessor` fixes this problem _unless_ the `SamplingTelemetryProcessor` gets in its way and "samples" the requests.

**Comment by cijothomas** [comment]
when you add application insights using AddApplicationInsightsTelemetry(), pass in disablesampling flag. Then add the sampling and your custom telemetryprocessor, in the order you want. (sampling **after** your processor) - This should mitigate your issue - please let me know if this helps.

**Comment by cijothomas** [comment]
Yea the flags are the only way currently to NOT have a predefined processor. No API now to 'remove' a processor.
Please let us know if you face issues.

**Comment by EnCey** [comment]
So I tried the flag workaround like this in `WebHostBuilder.ConfigureServices`: 

```cs
var aiOptions = new ApplicationInsightsServiceOptions {
    InstrumentationKey = coreConfig.InstrumentationKey,
    EnableAdaptiveSampling = false
};
services.AddApplicationInsightsTelemetry(aiOptions);
services.AddApplicationInsightsTelemetryProcessor(myProcessorType);
```

and as by this [wiki page](https://github.com/Microsoft/ApplicationInsights-aspnetcore/wiki/Telemetry-Processors:-Sampling-and-Metrics-Stream) in `Startup.Configure`:

```cs
var telemetryConfig = app.ApplicationServices.GetService<TelemetryConfiguration>();
telemetryConfig.TelemetryProcessorChainBuilder.UseAdaptiveSampling(5);
telemetryConfig.TelemetryProcessorChainBuilder.Build();
```

However, when I look at the `telemetryConfig` in a debug build, it still contains 5 TelemetryProcessors, including the 2 default AdaptiveSampling processors. After the call to `UseAdaptiveSampling` it contains 3. My custom processor is nowhere to be seen.

I have to say, the whole Application Insights configuration story in ASP.Net Core is an absolute nightmare. Add initializers here as singleton before `AddApplicationInsightsTelemetry`, processors afterward with a special extension method, turn off some stuff with a flag, get a `TelemetryConfiguration` in `Startup` and mess around with that.... If I could just get a `TelemetryConfiguration` at one point and be allowed to configure it I would be happy beyond reason.

**Comment by EnCey** [comment]
Alright after reading the source code I discovered that it's the `DefaultTelemetrySink` that's actually being configured. It contains my custom processor and the `AdaptiveSampling` processors are absent. 

I changed my code in `Startup.Configure` to also target the `DefaultTelemetrySink`:

```
var telemetryConfig = app.ApplicationServices.GetService<TelemetryConfiguration>();
var telemetryProcessorBuilder = telemetryConfig.DefaultTelemetrySink.TelemetryProcessorChainBuilder;
telemetryProcessorBuilder.UseAdaptiveSampling(5);
telemetryProcessorBuilder.Build();
```

which seems to do the trick. The `TelemetryConfiguration` still contains the AdaptiveSampling processors and not my custom one, but apparently it hasn't any effect anyway.
